tests/run/test_run.py

```diff
@@ -378,3 +382,50 @@ def test_verify_tag_in_output_for_relogged_artifact(setup_project):
 
     assert "v3" in output_uri, "Expected 'v3' tag in output_uri"
     assert "v3" in outputs_uri, "Expected 'v3' tag in outputs_uri"
+
+
+def test_code_to_function_invalid_handler_name_for_nuclio_mlrun_run_kind():
+    with pytest.raises(
+        mlrun.errors.MLRunInvalidArgumentError,
+        match=ERROR_MSG_INVALID_HANDLER_NAME,
+    ):
+        mlrun.code_to_function(
+            filename=f"{assets_path}/fail.py",
+            name="nuclio-mlrun",
+            image="mlrun/mlrun",
+            kind="nuclio:mlrun",
+            handler="handler",
+        )
+
+
+def test_nuclio_mlrun_invalid_handler_name():
+    function = mlrun.code_to_function(
+        filename=f"{assets_path}/fail.py",
+        name="nuclio-mlrun",
+        image="mlrun/mlrun",
+        kind="nuclio:mlrun",
+    )
+
+    with pytest.raises(
+        mlrun.errors.MLRunInvalidArgumentError,
+        match=ERROR_MSG_INVALID_HANDLER_NAME,
+    ):
+        function.run(handler="handler")
+
+
+def test_nuclio_mlrun_invalid_handler_signature():
+    def handler(context):
+        context.log_result("accuracy", 16)
+
+    function = mlrun.code_to_function(
+        filename=f"{assets_path}/fail.py",
+        name="nuclio-mlrun",
+        image="mlrun/mlrun",
+        kind="nuclio:mlrun",
+    )
+
+    with pytest.raises(
+        mlrun.errors.MLRunInvalidArgumentError,
+        match=ERROR_MSG_INVALID_HANDLER_NAME,
+    ):
+        function.run(handler=handler)
```

Comment:
We cannot run Nuclio function - it can only be deployed
```suggestion
        function.deploy()
```

(and same above)

Reply:
it is possible to run nuclio functions when subkind is mlrun. It essentially invokes the function

Reply:
@alonmr I just tried to run function:

```
function = project.set_function(
    func="func.py",
    name="nuclio-mlrun-2",
    image="mlrun/mlrun",
    kind="nuclio:mlrun",
    handler="func",
)
function.run()
```

and it doesn't work...

Reply:
I think the flow is
1. set_function
2. deploy
3. run

Reply:
Of course you need to deploy 1st because you need a function to be able to invoke it 🙂 
Specifying a handler should work with mlrun subkind. If it doesn't work might be a bug. I doubt anyone tried it lately

Reply:
We have discussed with @moranbental all the scenarios. As per our discussion, this test doesn't make sense and will be removed and replaced with another one (with 2 functions to invoke - one with correct name and another one called `handler`).

Reply:
@rokatyy I am currently checking the file code to see if there is a function called `handler` aside from the footer function.